### PR TITLE
[PIL-669][OK] - Updated regex

### PR DESCRIPTION
**Pull request by olakunlekuye2**

### CODE REVIEW

File: app/forms/NfmContactNameFormProvider.scala [hunk 1]
```diff
@@ -22,10 +22,11 @@ import play.api.data.Form
 import javax.inject.Inject
 
 class NfmContactNameFormProvider @Inject() extends Mappings {
-
+  private val scriptInjectionRegex = """[^<>]+"""
```
Comment (jassalrichy) [inline app/forms/NfmContactNameFormProvider.scala:25]: Would be better to use a valid email format regex ?
for example:
  val emailRegex = """^(?!\.)("([^"\r\\]|\\["\r\\])*"|([-a-zA-Z0-9!#$%&'*+/=?^_`{|}~]|(?<!\.)\.)*)(?<!\.)@[a-zA-Z0-9][\w\.-]*[a-zA-Z0-9]\.[a-zA-Z][a-zA-Z\.]*[a-zA-Z]$"""

Comment (jassalrichy) [reply app/forms/NfmContactNameFormProvider.scala:25]: Also useful to make say a Validation object as part of the forms package to include all common regex ?

File: app/forms/NfmEmailAddressFormProvider.scala [hunk 2]
```diff
@@ -22,13 +22,15 @@ import forms.mappings.Mappings
 import play.api.data.Form
 
 class NfmEmailAddressFormProvider @Inject() extends Mappings {
-  val maxLength  = 132
-  val emailRegex = s"^(.{1,$maxLength})@([a-zA-Z0-9.-]+)\\.([a-zA-Z]{2,6})$$"
+  val maxLength            = 132
+  val emailRegex           = s"^(.{1,$maxLength})@([a-zA-Z0-9.-]+)\\.([a-zA-Z]{2,6})$$"
+  val scriptInjectionRegex = """[<>\"'`]+""".r
```
Comment (jassalrichy) [inline app/forms/NfmEmailAddressFormProvider.scala:27]: See above

File: app/forms/NfmContactNameFormProvider.scala (hunk 1 above)
Comment (olakunlekuye2) [reply app/forms/NfmContactNameFormProvider.scala:25]: I'm renaming the val to nameRegex   as suggested  but the refactoring can go into tech debt

File: app/forms/NfmEmailAddressFormProvider.scala (hunk 2 above)
Comment (olakunlekuye2) [reply app/forms/NfmEmailAddressFormProvider.scala:27]: I'll play around with email regex but on first pass it wont accept ola@abc.com

Comment (olakunlekuye2) [reply app/forms/NfmEmailAddressFormProvider.scala:27]: I'm just renaming the val

Comment (jassalrichy) [reply app/forms/NfmEmailAddressFormProvider.scala]: I've checked this regex using [regex.com](https://regex101.com/) (with flavour set to ECMAScript) with ola@abc.com and works fine for me.

File: conf/messages.en [hunk 3]
```diff
@@ -387,6 +387,7 @@ nfmContactName.checkYourAnswersLabel = Contact name
 nfmContactName.error.required = You need to enter the name of the person or team we should contact from the nominated filing member
 nfmContactName.error.length = The name of the contact person or team should be 105 characters or less
 nfmContactName.change.hidden =  filing member contact name
+nfmContactName.error.scriptinjection = The name entered contains invalid characters. Please remove any special characters like < or >
```
Comment (jassalrichy) [inline conf/messages.en:390]: This message is now redundant and can be removed?

File: conf/messages.en [hunk 4]
```diff
@@ -397,6 +398,7 @@ nfmEmailAddress.error.required = You need to enter the email address for {0}
 nfmEmailAddress.error.length = The email address should be 132 characters or less
 contactEmailAddress.error.format = Enter an email address in the correct format, like name@example.com
 nfmEmailAddress.change.hidden = Changing filing member contact email
+nfmEmailAddress.error.scriptinjection = The email entered contains invalid characters. Please remove any special characters like < or >
```
Comment (jassalrichy) [inline conf/messages.en:401]: This message is now redundant and can be removed?

File: app/forms/NfmContactNameFormProvider.scala [hunk 5]
```diff
@@ -22,10 +22,11 @@ import play.api.data.Form
 import javax.inject.Inject
 
 class NfmContactNameFormProvider @Inject() extends Mappings {
-
+  private val nameRegex = """[^<>]+"""
```
Comment (jassalrichy) [inline app/forms/NfmContactNameFormProvider.scala:25]: Ideally we should create a Object for all the regex values as these are used in multiple places.

Comment (jassalrichy) [reply app/forms/NfmContactNameFormProvider.scala:25]: We should use the correct regex for Name values: "^[A-Za-z0-9 ,.()/&'-]*$"

File: app/forms/NfmContactNameFormProvider.scala [hunk 6]
```diff
@@ -22,10 +22,11 @@ import play.api.data.Form
...
   def apply(): Form[String] =
     Form(
       "value" -> text("nfmContactName.error.required")
         .verifying(maxLength(105, "nfmContactName.error.length"))
+        .verifying(regexp(nameRegex, "nfmContactName.error.scriptinjection"))
     )
```
Comment (jassalrichy) [inline app/forms/NfmContactNameFormProvider.scala:31]: Perhaps rename scriptinjection to invalid

File: app/forms/UpeContactNameFormProvider.scala [hunk 7]
```diff
@@ -23,11 +23,12 @@ import javax.inject.Inject
 
 class UpeContactNameFormProvider @Inject() extends Mappings {
 
-  val max = 200
-
+  val max               = 200
+  private val nameRegex = """[^<>]+"""
```
Comment (jassalrichy) [inline app/forms/UpeContactNameFormProvider.scala:27]: See comment above

File: app/forms/UpeContactNameFormProvider.scala [hunk 8]
```diff
@@ -23,11 +23,12 @@ import javax.inject.Inject
...
   def apply(): Form[String] =
     Form(
       "value" -> text("upe-input-business-name.error.required")
         .verifying(maxLength(max, "upe-input-business-name.error.length"))
+        .verifying(regexp(nameRegex, "nfmContactName.error.scriptinjection"))
```
Comment (jassalrichy) [inline app/forms/UpeContactNameFormProvider.scala:32]: see comment above

File: conf/messages.en (hunk 3 above)
Comment (jassalrichy) [inline conf/messages.en:390]: see comment above

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**